auction-app-backend/src/main/java/com/example/auctionapp/repository/ProductRepository.java

```diff
@@ -32,4 +32,15 @@ Page<ProductEntity> findProductEntityByUserEntity_UserIdAndAndStatus(final UUID
     @Modifying
     @Query("UPDATE ProductEntity p SET p.status = 'INACTIVE' WHERE p.status = 'ACTIVE' AND p.endDate <= :now")
     int updateProductStatus(final LocalDateTime now);
+
+    List<ProductEntity> findProductEntitiesByUserEntity_UserId(final UUID userId);
+
+    @Query(value = "SELECT p FROM ProductEntity p WHERE p.categoryEntity.categoryId IN " +
+            "(SELECT c.categoryId FROM CategoryEntity c " +
+            "WHERE c.parentCategory.categoryId = :categoryId OR c.categoryId = :categoryId) AND p.status = 'ACTIVE' " +
+            "ORDER BY p.bidsCount DESC, p.endDate ASC")
+    List<ProductEntity> findTopPopularProductEntitiesByCategoryId(final UUID categoryId, final Pageable pageable);
```

Comment:
Check if this can be done with LIMIT

Reply:
i tired to do it with limit but i kept getting query validation error, so i checked it out and i guess jpql does not support the use of limit. i left pageable but i tried to shorten the service methods instead.